frontend/app/routes/$lang+/_protected+/personal-information+/preferred-language+/edit.tsx

```diff
@@ -36,19 +38,20 @@ export const meta: MetaFunction<typeof loader> = mergeMeta(({ data }) => {
   return data ? getTitleMetaTags(data.meta.title) : [];
 });
 
-export async function loader({ context: { session }, request }: LoaderFunctionArgs) {
+export async function loader({ context: { session }, params, request }: LoaderFunctionArgs) {
   featureEnabled('edit-personal-info');
   const instrumentationService = getInstrumentationService();
   const lookupService = getLookupService();
   const raoidcService = await getRaoidcService();
-  const userService = getUserService();
 
   await raoidcService.handleSessionValidation(request, session);
 
-  const userId = await userService.getUserId();
-  const userInfo = await userService.getUserInfo(userId);
+  const userInfoToken: UserinfoToken = session.get('userInfoToken');
+  const personalInformationRouteHelpers = getPersonalInformationRouteHelpers();
+  const personalInformation = await personalInformationRouteHelpers.getPersonalInformation(userInfoToken, params, request, session);
+  const preferredLanguageId = personalInformation.preferredLanguageId;
 
-  if (!userInfo) {
+  if (!preferredLanguageId) {
     instrumentationService.countHttpStatus('preferred-language.edit', 404);
     throw new Response(null, { status: 404 });
   }
```

Comment:
No need to throw a 404 here. The user may not have a preferred language initially set and may want to select one for the first time.

Reply:
removed and added validation